### PR TITLE
Bug/vspc 207

### PR DESCRIPTION
**Pull request by swetalina-prusty25**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 
Staff side: There should be a search for images

...Put description here...
When going to "Images" on the staff side, right now it shows a table of all images. There should be a search to filter images. The search should search filename, name, and description for the given search term.
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/IImageService.java [hunk 1]
```diff
@@ -35,6 +36,7 @@ public interface IImageService {
 
     void removeCategory(IVSImage image, ImageCategory category);
 
-    List<IVSImage> findByFilenameOrNameContains(String searchTerm);
+    List <IVSImage> findByFilenameOrNameContains(String searchTerm);
 
-}
\ No newline at end of file
+    List <IVSImage> getImagesForPagination(int pageNo, ImageCategory category, String searchTerm);
+}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/IImageService.java:42]: there should be no blanks before `<...`

**Review by jdamerow — CHANGES_REQUESTED** [review]
Please remove all unnecessary formatting changes and reopen

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/IImageService.java (hunk 1 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/IImageService.java:42]: removed

**Comment by jdamerow** [comment]
> there should be no blanks before <...

There are still many places where this is the case.

**Comment by swetalina-prusty25** [comment]
Removed extra spaces befor <>

**Review by jdamerow — CHANGES_REQUESTED** [review]
there are still a bunch of unnecessary blanks before `<`

**Comment by swetalina-prusty25** [comment]
i misunderstood with blank spaces before < with blank spaces between < and >. Now i have removed those.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java [hunk 2]
```diff
@@ -126,23 +124,23 @@ public List<IVSImage> getImages(int pageNo) {
...
-        if(category==null) {
+        Page <VSImage> images;
+        if (category == null) {
             images = imageRepo.findAll(sortByRequestedField);
         } else {
             images = imageRepo.findByCategories(sortByRequestedField, category);
         }
-        List<IVSImage> results = new ArrayList<>();
-        if(images != null) {
+        List <IVSImage> results = new ArrayList < > ();
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:137]: there are multiple instance of blanks before `<...>` in here

**Review by jdamerow — CHANGES_REQUESTED** [review]
there are still a lot of instances of blanks before `<...>`. please don't reopen until this is fixed.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java (hunk 2 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:137]: i have removed them. In ImageService.java there were bunch of spaces.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package edu.asu.diging.vspace.web.staff.api;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:1]: indentation

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @RequestMapping("/staff/images/search/full")
+	public String imageSearchFull(Model model,
+	@RequestParam(value = "searchText", required = false) String searchTerm) {
+    	return String.format("redirect:/staff/images/search/full/1?searchText=%s",(searchTerm==null?"":searchTerm));
+    }
+    
+    @RequestMapping("/staff/images/search/full/{page}")
+	public String imageSeacrhDescription(@PathVariable(required = false) String page,
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:40]: if `page` is not required, why do you have an extra controller method?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java (hunk 3 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:1]: fixed

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java (hunk 4 above)
Comment (swetalina-prusty25) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:40]: have used only one controller method that is the second one. Removed the first controller method as it was not necessary and made the required changes

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java [hunk 5]
```diff
@@ -250,4 +248,28 @@ public void removeCategory(IVSImage image, ImageCategory category) {
         image.getCategories().remove(category);
         imageRepo.save((VSImage) image);
     }
+    
+    /**
+     * Method to return the requested images
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:253]: what does this mean? what does this method do?

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java [hunk 6]
```diff
@@ -250,4 +248,28 @@ public void removeCategory(IVSImage image, ImageCategory category) {
...
+     * 
+     * @param pageNo.    if pageNo<1, 1st page is returned, if pageNo>total
+     *                   pages,last page is returned
+     * @param searchTerm - This is the search string which is being searched.
+     * @param images     - This is the result list of search images.
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List<IVSImage> getImagesForPagination(int pageNo, ImageCategory category, String searchTerm) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:262]: bad method name. Should be as self-explanatory as possible.

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java [hunk 7]
```diff
@@ -250,4 +248,28 @@ public void removeCategory(IVSImage image, ImageCategory category) {
...
+        pageNo = validatePageNumber(pageNo, category);
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize, sortingParameters);
+        String likeSearchTerm = "%" + searchTerm + "%";
+        Page<VSImage> result = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(sortByRequestedField,
+            likeSearchTerm, likeSearchTerm, likeSearchTerm);
+        List<IVSImage> imageResults = new ArrayList<>();
+        if (result != null) {
+            result.getContent().forEach(i -> imageResults.add(i));
+        }
+        return imageResults;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java:273]: can't you just return `result.getContent()`?

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImageSearchApiController.java
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImageSearchApiController.java:1]: there are formatting changes that do not comply with the code style we use.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java (hunk 3 above)
Comment (jdamerow) [reply vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:1]: no it's not

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    @Autowired
+	private IImageService imageService;
+
+    @Autowired
+	private ISpaceManager spaceManager;
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+  
+    @RequestMapping("/staff/images/search/full/{page}")
+	public String imageSeacrhDescription(@PathVariable(required = false) String page,
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java:35]: there is s spelling mistake.

**Review by jdamerow — CHANGES_REQUESTED** [review]
I'm going to stop reviewing here. Please adhere to our code style, check for spelling mistakes, unnecessary file changes, indentation, etc.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- there needs to be a way to clear the search results
- the pagination should be adjusted to search results (right now it just stays but moves to empty page)
- if I search again after I searched, I get an error

**Pull request closed without merging**